# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.cpp
# Commit: d529ccb6638b
# Full Hash: d529ccb6638b92890550fc5c8a13092b6b72e450
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-30 10:00:42
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P13 Avoid ODA message being processed when the channel is already failed, r=mayhemer,kershaw,necko-reviewers
#   
#   One line fix with a bunch of explanation.
#   
#   When we perform networking on the socket process, OnDataAvailable could be sent directly from socket process to child process.
# ==============================================================================

diff -r cad764eb15a4 -r d529ccb6638b netwerk/protocol/http/HttpChannelChild.cpp
--- a/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 30 00:34:32 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.cpp	Tue Jun 30 00:34:39 2020 +0000
@@ -775,7 +775,7 @@
     return;
   }
 
-  if (mCanceled) {
+  if (mCanceled || NS_FAILED(mStatus)) {
     return;
   }
 
